### PR TITLE
Raffle history

### PR DESCRIPTION
**Pull request by andkoval**

### CODE REVIEW

File: src/pages/Raffle/components/RaffleDate.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            setTitle(`ended ${end.toRelative()}`);
+        }
+    };
+
+    if(!title) return null;
+
+    return (
+        <div
+            className={classNames(classes.title, type)}
+            // style={{ color: type === 'live' ? '#ff0c0c' : type === 'upcoming' ? '#1fd71f' : '#979797' }}
```
Comment (dmutrokot) [inline src/pages/Raffle/components/RaffleDate.js:44]: please remove if not needed

File: src/pages/Raffle/components/RaffleDate.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useInterval(() => {
+        renderTitle();
+    }, 1000);
+
+    const renderTitle = () => {
+        let local = DateTime.local();
```
Comment (dmutrokot) [inline src/pages/Raffle/components/RaffleDate.js:24]: please use const instead and line below also

File: src/contexts/RaffleContext.js [hunk 3]
```diff
@@ -100,28 +117,49 @@ const RaffleContextProvider = (props) => {
...
     const countChances = (value, entered, items) => {
-        return value / entered * items;
+        let supply = raffle.endDate - DateTime.local() < 0 ? entered  : +entered + +value;
+
+        return value / supply * items;
     }
 
-    const formatChance = (chance, items) => {
-        let percentage = (chance * 100).toFixed(1);
+    const countWearablesChances = (ticket) => {
```
Comment (dmutrokot) [inline src/contexts/RaffleContext.js:125]: please replace let with const in the method

File: src/contexts/RaffleContext.js [hunk 4]
```diff
@@ -100,28 +117,49 @@ const RaffleContextProvider = (props) => {
     };
 
     const countChances = (value, entered, items) => {
-        return value / entered * items;
+        let supply = raffle.endDate - DateTime.local() < 0 ? entered  : +entered + +value;
```
Comment (dmutrokot) [inline src/contexts/RaffleContext.js:120]: please use const

File: src/api/thegraph.js [hunk 5]
```diff
@@ -262,6 +270,25 @@ export default {
         });
     },
 
+    async getRaffleWins(address, raffle) {
+        return await this.getRaffleData(raffleWinsQuery(address.toLowerCase())).then((response) => {
+            let data = [];
```
Comment (dmutrokot) [inline src/api/thegraph.js:275]: please use const instead

**Review by an-uhryn — APPROVED** [review]

**Pull request merged**